as560x/as560x.go

```diff
@@ -0,0 +1,193 @@
+func (d *BaseDevice) ReadRegister(address uint8) (uint16, error) {
+	reg, ok := d.registers[address]
+	if !ok {
+		return 0, errors.New("Register not found")
+	}
+	return reg.read(d.bus, d.address)
+}
+
+// WriteRegister writes the given value for the given register to the AS560x device via I2C
+func (d *BaseDevice) WriteRegister(address uint8, value uint16) error {
+	reg, ok := d.registers[address]
+	if !ok {
+		return errors.New("Register not found")
+	}
+	return reg.write(d.bus, d.address, value)
+}
+
+// GetZeroPosition returns the 'zero position' (ZPOS) in various units
+func (d *BaseDevice) GetZeroPosition(units AngleUnit) (uint16, float32, error) {
+	zpos, err := d.ReadRegister(ZPOS)
+	if nil != err {
+		return 0, 0.0, err
+	}
+	// Convert to requested units
+	i, f := convertFromNativeAngle(zpos, NATIVE_ANGLE_RANGE, units)
+	return i, f, nil
+}
+
+// SetZeroPosition sets the 'zero position' (ZPOS) in various units
+func (d *BaseDevice) SetZeroPosition(zpos float32, units AngleUnit) error {
+	return d.WriteRegister(ZPOS, convertToNativeAngle(zpos, units))
+}
+
+// RawAngle reads the (unscaled & unadjusted) RAW_ANGLE register in various units
+func (d *BaseDevice) RawAngle(units AngleUnit) (uint16, float32, error) {
+	angle, err := d.ReadRegister(RAW_ANGLE)
+	if nil != err {
+		return 0, 0.0, err
+	}
+	// Convert to requested units
+	i, f := convertFromNativeAngle(angle, NATIVE_ANGLE_RANGE, units)
+	return i, f, nil
+}
+
+// Angle reads the (scaled & adjusted) ANGLE register in various units
+func (d *BaseDevice) Angle(units AngleUnit) (uint16, float32, error) {
+	// ZPOS enables setting the 'zero position' of the device to any RAW_ANGLE value
+	// ANGLE is RAW_ANGLE adjusted relative to ZPOS.
+	angle, err := d.ReadRegister(ANGLE)
+	if nil != err {
+		return 0, 0.0, err
+	}
+	// Convert to requested units
+	i, f := convertFromNativeAngle(angle, d.maxAngle, units)
+	return i, f, nil
+}
+
+// MagnetStatus reads the STATUS register and reports magnet position characteristics
+func (d *BaseDevice) MagnetStatus() (detected bool, strength MagnetStrength, err error) {
+	status, err := d.ReadRegister(STATUS)
+	if nil != err {
+		return false, MagnetOk, err
+	}
+	detected = (status & STATUS_MD) != 0
+	strength = MagnetOk
+	if (status & STATUS_ML) != 0 {
+		strength = MagnetTooWeak
+	} else if (status & STATUS_MH) != 0 {
+		strength = MagnetTooStrong
+	}
+	return
+}
+
+// Burn is a convenience method to program the device permanently by writing to the BURN register (limited number of times use!)
+func (d *BaseDevice) Burn(burnCmd BURN_CMD) error {
+	if BURN_ANGLE == burnCmd {
+		// BURN_ANGLE can only be executed up to 3 times.
+		// We can check this in advance by reading ZMCO before writing to the BURN register.
+		numBurns, err := d.ReadRegister(ZMCO)
+		if nil != err {
+			return err
+		}
+		if numBurns >= BURN_ANGLE_COUNT_MAX {
+			// We're outta BURNs :(
+			return errors.New("Max BURN_ANGLE limit reached")
```

Comment:
Better to have this error defined as package var

```
var errMaxBURNAngle = errors.New("Max BURN_ANGLE limit reached")
```

And then in this function

```go
return errMaxBURNAngle
```

It avoids extra heap allocations.